### ISSUE TITLE
Parsing ReadableMaps is broken on RN 0.54

### ISSUE THREAD

**Issue by ebouJ** [wix/react-native-navigation#3313]
### Issue Description

The app crash on Android but works fine on IOS.
<img width="285" alt="screen shot 2018-06-06 at 12 29 14 pm" src="https://user-images.githubusercontent.com/22877561/41054864-9886a226-698d-11e8-87a5-d3f4a79f4cbd.png">

I checked the code that bridges and set the setRoot react method here and it seems like the problem is on line 60

<img width="995" alt="screen shot 2018-06-06 at 1 31 30 pm" src="https://user-images.githubusercontent.com/22877561/41054981-fc6b3db0-698d-11e8-8f1b-c9f663a1e9c2.png">

MainActivity.java
```java
        import com.reactnativenavigation.*;

        public class MainActivity extends NavigationActivity {
        }
```

MainApplication.java
```java
package com.cityworksmobile;

import android.app.Application;

import com.facebook.react.ReactApplication;
import com.microsoft.appcenter.reactnative.crashes.AppCenterReactNativeCrashesPackage;
import com.microsoft.appcenter.reactnative.analytics.AppCenterReactNativeAnalyticsPackage;
import com.microsoft.appcenter.reactnative.appcenter.AppCenterReactNativePackage;
import com.oblador.keychain.KeychainPackage;
import com.AlexanderZaytsev.RNI18n.RNI18nPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;
import com.reactnativenavigation.NavigationApplication;
import com.reactnativenavigation.react.NavigationReactNativeHost;
import android.support.annotation.*;
import java.util.*;



public class MainApplication extends NavigationApplication {

   @Override
    public boolean isDebug() {
        return BuildConfig.DEBUG;
    }

        @Override
    protected ReactNativeHost createReactNativeHost() {
        return new NavigationReactNativeHost(this) {
            @Override
            protected String getJSMainModuleName() {
                return "index";
            }
        };
    }
   
    protected List<ReactPackage> getPackages() {
        // Add additional packages you require here
        // No need to add RnnPackage and MainReactPackage
        return Arrays.<ReactPackage>asList(
            new AppCenterReactNativeCrashesPackage(MainApplication.this, getResources().getString(R.string.appCenterCrashes_whenToSendCrashes)),
            new AppCenterReactNativeAnalyticsPackage(MainApplication.this, getResources().getString(R.string.appCenterAnalytics_whenToEnableAnalytics)),
            new AppCenterReactNativePackage(MainApplication.this),
            new KeychainPackage(),
            new RNI18nPackage()
        );
    }
     
    @Nullable
    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
        return getPackages();
    }
}

```

build.gradle
```
apply plugin: "com.android.application"

import com.android.build.OutputFile

/**
 * The react.gradle file registers a task for each build variant (e.g. bundleDebugJsAndAssets
 * and bundleReleaseJsAndAssets).
 * These basically call `react-native bundle` with the correct arguments during the Android build
 * cycle. By default, bundleDebugJsAndAssets is skipped, as in debug/dev mode we prefer to load the
 * bundle directly from the development server. Below you can see all the possible configurations
 * and their defaults. If you decide to add a configuration block, make sure to add it before the
 * `apply from: "../../node_modules/react-native/react.gradle"` line.
 *
 * project.ext.react = [
 *   // the name of the generated asset file containing your JS bundle
 *   bundleAssetName: "index.android.bundle",
 *
 *   // the entry file for bundle generation
 *   entryFile: "index.android.js",
 *
 *   // whether to bundle JS and assets in debug mode
 *   bundleInDebug: false,
 *
 *   // whether to bundle JS and assets in release mode
 *   bundleInRelease: true,
 *
 *   // whether to bundle JS and assets in another build variant (if configured).
 *   // See http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Build-Variants
 *   // The configuration property can be in the following formats
 *   //         'bundleIn${productFlavor}${buildType}'
 *   //         'bundleIn${buildType}'
 *   // bundleInFreeDebug: true,
 *   // bundleInPaidRelease: true,
 *   // bundleInBeta: true,
 *
 *   // whether to disable dev mode in custom build variants (by default only disabled in release)
 *   // for example: to disable dev mode in the staging build type (if configured)
 *   devDisabledInStaging: true,
 *   // The configuration property can be in the following formats
 *   //         'devDisabledIn${productFlavor}${buildType}'
 *   //         'devDisabledIn${buildType}'
 *
 *   // the root of your project, i.e. where "package.json" lives
 *   root: "../../",
 *
 *   // where to put the JS bundle asset in debug mode
 *   jsBundleDirDebug: "$buildDir/intermediates/assets/debug",
 *
 *   // where to put the JS bundle asset in release mode
 *   jsBundleDirRelease: "$buildDir/intermediates/assets/release",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in debug mode
 *   resourcesDirDebug: "$buildDir/intermediates/res/merged/debug",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in release mode
 *   resourcesDirRelease: "$buildDir/intermediates/res/merged/release",
 *
 *   // by default the gradle tasks are skipped if none of the JS files or assets change; this means
 *   // that we don't look at files in android/ or ios/ to determine whether the tasks are up to
 *   // date; if you have any other folders that you want to ignore for performance reasons (gradle
 *   // indexes the entire tree), add them here. Alternatively, if you have JS files in android/
 *   // for example, you might want to remove it from here.
 *   inputExcludes: ["android/**", "ios/**"],
 *
 *   // override which node gets called and with what additional arguments
 *   nodeExecutableAndArgs: ["node"],
 *
 *   // supply additional arguments to the packager
 *   extraPackagerArgs: []
 * ]
 */

project.ext.react = [
    entryFile: "index.js"
]

apply from: "../../node_modules/react-native/react.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion 26
    buildToolsVersion '27.0.3'


    defaultConfig {
        applicationId "xxxxxxxxxxx"
        minSdkVersion 19
        targetSdkVersion 25
        versionCode 1
        versionName "1.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
        //multiDexEnabled = true
    }
      compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    // added that to resolved compatitibility issues
    configurations.all {
        resolutionStrategy.eachDependency { DependencyResolveDetails details ->
            def requested = details.requested
            if (requested.group == 'com.android.support' && details.requested.name != 'multidex') {
                details.useVersion '27.0.+'
            }
        }
    }
    //implementation 'com.android.support:multidex:1.0.1'
    implementation project(':appcenter-crashes')
    implementation project(':appcenter-analytics')
    implementation project(':appcenter')
    implementation project(':react-native-keychain')
    implementation project(':react-native-i18n')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation project(':react-native-navigation')



}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}


```

### Steps to Reproduce / Code Snippets / Screenshots

[FILL THIS OUT - It will be extremely helpful]

---
### Environment
* React Native Navigation version: 2.0.2324
* React Native version: 0.54.0
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Android Device/Emulator

**Comment by yogevbd** [comment]
@ebouJ Can you please post your `setRoot` code?

**Comment by ebouJ** [comment]
Sure here is the code @yogevbd 
```
import { Navigation } from "react-native-navigation"
import registerScreens from "./register-screens"

/* -- We don't need this for now
if (Platform.OS === "android") {
  alert = title => {
    Navigation.showOverlay({
      component: {
        name: "", // <= 
        passProps: {
          title,
        },
        options: {
          overlay: {
            interceptTouchOutside: true,
          },
        },
      },
    })
  }
}
*/

export default function start() {
  registerScreens()
  Navigation.events().registerAppLaunchedListener(() => {
    Navigation.setRoot({
      root: {
        stack: {
          id: "ONBOARDING",
          children: [
            {
              component: {
                name: "navigation.OnboardingScreen",
              },
            },
          ],
        },
      },
    })
  })
}
```

**Comment by guyca** [comment]
Hey @ebouJ 
Your code seems fine and the following identical layout works in the playground app:

```js
    Navigation.setRoot({
      root: {
        stack: {
          id: 'TEST',
          children: [
            {
              component: {
                name: 'navigation.playground.WelcomeScreen'
              }
            }
          ]
        }
      }
    })
```

If you think this is a bug in RNN, please PR a failing detox test.

**Comment by ebouJ** [comment]
@guyca i disagree. This is still an issue because I could not even run the playground app on Android

**Comment by guyca** [comment]
The playground app runs fine for us. What error are you getting when running the playground app?

**Comment by ebouJ** [comment]
There is no index.android.js in the repo to test it on Android. I tried adding it but it never works. How are you guys running the playground project? Also is the fragment classes in the playground example needed to run it on Android in v2?

**Comment by ebouJ** [comment]
@guyca can you please update the documentation please after I downgraded my react-native version to 0.51.1 everything works fine.

**Comment by guyca** [comment]
@ebouJ 
I realised this is indeed a regression introduced in RN. I'm pushing a fix now, it should be published next week.

**Comment by harrisrobin** [comment]
Having this issue as well.

**Comment by guyca** [comment]
Seems like this is fixed in RN 0.55.4
